File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/impl/supplier/SwlRsaAsymmetricEncryptorSupplier.java
package i2f.swl.impl.supplier;

import i2f.swl.impl.SwlRsaAsymmetricEncryptor;
import i2f.swl.std.ISwlAsymmetricEncryptor;
import i2f.swl.std.supplier.ISwlAsymmetricEncryptorSupplier;

/**
 * @author Ice2Faith
 * @date 2024/7/10 18:59
 * @desc
 */
public class SwlRsaAsymmetricEncryptorSupplier implements ISwlAsymmetricEncryptorSupplier {
    @Override
    public ISwlAsymmetricEncryptor get() {
        return new SwlRsaAsymmetricEncryptor();
    }
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-swl/core/impl/supplier/SwlRsaAsymmetricEncryptorSupplier.js
import SwlRsaAsymmetricEncryptor from "../SwlRsaAsymmetricEncryptor";

/**
 * @return {SwlRsaAsymmetricEncryptorSupplier}
 * @implements ISwlAsymmetricEncryptorSupplier
 */
function SwlRsaAsymmetricEncryptorSupplier() {
    return {
        /**
         * @return {ISwlAsymmetricEncryptor}
         */
        get() {
            return SwlRsaAsymmetricEncryptor()
        }
    }
}

export default SwlRsaAsymmetricEncryptorSupplier
